### PR TITLE
Format multiline doc strings to match style of other help messages

### PR DESCRIPTION
**Pull request by bambu**
Changed cmd2 do_cmd to dedent doc strings using `pydoc.getdoc` in order for the output of multi-line doc strings to look similar to argparse or a single line doc string. It continues prioritizing `help_func` functions over doc strings.

Before patch:
```
(cmd2) ➜  cmd2 git:(docstr_fmt) ✗ python examples/help_categories.py
(Cmd) help findleakers
Find Leakers command
(Cmd) help sslconnectorciphers

        SSL Connector Ciphers command is an example of a command that contains
        multiple lines of help information for the user. Each line of help in a
        contiguous set of lines will be printed and aligned in the verbose output
        provided with 'help --verbose'

        This is after a blank line and won't de displayed in the verbose help
```

After patch:
```shell
(cmd2) ➜  cmd2 git:(docstr_fmt) ✗ python examples/help_categories.py
(Cmd) help findleakers
Find Leakers command
(Cmd) help sslconnectorciphers
SSL Connector Ciphers command is an example of a command that contains
multiple lines of help information for the user. Each line of help in a
contiguous set of lines will be printed and aligned in the verbose output
provided with 'help --verbose'

This is after a blank line and won't de displayed in the verbose help
```

### CODE REVIEW

**Comment by tleonhardt** [comment]
@bambu Thanks for the PR.  Can you please fix the `flake8` error(s)?  

@kmvanbrunt @anselor Do either of you have an opinion about the changes in this PR?  It changes the way help is displayed, but it isn't clear to me if the change makes things better or worse or if its just neautral.

**Comment by kmvanbrunt** [comment]
@tleonhardt I would like some time to check out the effects of this PR. Unfortunately I've been sick, so I'll take a look when I'm well.

**Comment by tleonhardt** [comment]
@kmvanbrunt I don’t think there is any rush for merging this in.  @bambu needs to fix the flake8 error(s) first anyways.  So take your time.

File: cmd2/cmd2.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import inspect
 import os
 import pickle
+import pydoc
```
Comment (tleonhardt) [inline cmd2/cmd2.py:37]: I've never heard of `pydoc` before.  Is it a built-in module?

Comment (anselor) [reply cmd2/cmd2.py:37]: I looked it up and it appears to be a built-in.

Comment (tleonhardt) [reply cmd2/cmd2.py:37]: Coming from a background in C/C++, the sheer volume of stuff built into Python's standard library continues to amaze me even after working with it for over a decade!

**Review by tleonhardt — APPROVED** [review]
To me, the change appears neutral - better in some ways and worse in others.  It doesn't seem to hurt anything.  I'm pretty ambivalent in regards to whether we merge this in or reject it. 

@kmvanbrunt @anselor @kotfu If any of you has an opinion, please weigh in.  There is no rush to merge this, so lets gather input from all interested in providing it.

**Comment by tleonhardt** [comment]
@bambu Thanks for fixing the minor flake8 error so the all of the continuous integration checks can pass.

**Comment by bambu** [comment]
@tleonhardt it is definitely a minor change. I primarily like the consistency of having the same amount of indentation whether you are using argparse or a docstring. IMHO it also looks odd when the docstring itself has some indentation. In these cases there are multiple levels of indentation and usually inconsistent spacing.

File: cmd2/cmd2.py [hunk 2]
```diff
@@ -3155,17 +3156,21 @@ def do_help(self, args: argparse.Namespace) -> None:
                 # Set end to blank so the help output matches how it looks when "command -h" is used
                 self.poutput(completer.format_help(tokens), end='')
 
+            # If there is a help func delegate to do_help
+            elif help_func:
```
Comment (kmvanbrunt) [inline cmd2/cmd2.py:3160]: I prefer more explicit checks for None.
`elif help_func is not None:`

File: cmd2/cmd2.py [hunk 3]
```diff
@@ -3155,17 +3156,21 @@ def do_help(self, args: argparse.Namespace) -> None:
...
+                super().do_help(args.command)
+
+            # If there's no help_func __doc__ then format and output it
+            elif func and func.__doc__:
```
Comment (kmvanbrunt) [inline cmd2/cmd2.py:3164]: Same here:
`elif func is not None and func.__doc__ is not None:`

**Review by kmvanbrunt — CHANGES_REQUESTED** [review]

**Review by kmvanbrunt — APPROVED** [review]

**Review by tleonhardt — APPROVED** [review]
Changes look good.

TravisCI appears to potentially broken for us today.  If its a temporary thing fine.  If not, we might need to look into our configuration and/or moving to GitHubActions.

**Pull request merged**